src/Network/AWS/ARN/States.hs

```diff
@@ -0,0 +1,46 @@
+{-# LANGUAGE DeriveAnyClass #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+module Network.AWS.ARN.States
+  ( StateMachine (..),
+    parseStateMachine,
+    _StateMachine,
+  )
+where
+
+import Data.Hashable (Hashable)
+import Data.Text (Text)
+import qualified Data.Text as T
+import GHC.Generics (Generic)
+import Lens.Micro.Pro (Prism', prism')
+
+-- $setup
+-- >>> :set -XOverloadedStrings
+-- >>> import Lens.Micro.Pro ((^?))
+
+-- | An AWS State Machine, made of its name.
+--
+-- >>> "stateMachine:orderProcessor" ^? _StateMachine
+-- Just (StateMachine {name = "orderProcessor"})
+--
+-- @since 0.3.3.0
+newtype StateMachine = StateMachine
+  { name :: Text
+  }
+  deriving (Eq, Ord, Hashable, Show, Generic)
+
+-- | @since 0.3.3.0
+parseStateMachine :: Text -> Maybe StateMachine
+parseStateMachine t = case T.splitOn ":" t of
+  ["stateMachine", nam] ->
+    Just (StateMachine nam)
+  _ -> Nothing
+
+-- | @since 0.3.3.0
+renderStateMachine :: StateMachine -> Text
+renderStateMachine (StateMachine nam) = "stateMachine:" <> nam
```

Comment:
I'm not _thrilled_ by this shortening but it's something we've been doing in other modules in this package so it's at least consistent (and it was probably my fault).

Reply:
Which other module uses the same shortening? All `Lambda.hs` uses `T.intercalate` and `S3.hs` uses `<> "/" <>`. I'm happy to change `stateMachine:` to `T.intercalate`  if you think that's better.